src/main/java/com/linecorp/armeria/client/routing/ServerGroupInvoker.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import com.linecorp.armeria.client.ClientCodec;
+import com.linecorp.armeria.client.ClientOptions;
+import com.linecorp.armeria.client.DecoratingRemoteInvoker;
+import com.linecorp.armeria.client.RemoteInvoker;
+import io.netty.channel.EventLoop;
+import io.netty.util.concurrent.Future;
+
+import java.lang.reflect.Method;
+import java.net.URI;
+
+public class ServerGroupInvoker extends DecoratingRemoteInvoker {
+    private String groupName;
+
+    /**
+     * Creates a new instance that decorates the specified {@link RemoteInvoker}.
+     *
+     * @param delegate
+     */
+    public ServerGroupInvoker(RemoteInvoker delegate,String groupName) {
+        super(delegate);
+        this.groupName = groupName;
+    }
+
+    @Override
+    public <T> Future<T> invoke(EventLoop eventLoop, URI uri,
+                                ClientOptions options,
+                                ClientCodec codec, Method method,
+                                Object[] args) throws Exception {
+        Node selectedNode = ServerGroupRegistryHub.selectNode(groupName);
+        uri = URI.create(uri.toString().replaceFirst(groupName, selectedNode.address()));
```

Comment:
@trustin Is it ok for client = newClient("www.google.com"); client.execute("www.microsoft.com") to work? This behavior surprises me - if it's ok, we need to document in the client builders that an authority in a uri will be ignored if the request has a different uri. But I feel like (without having done due diligence on the code) HTTP2-related mechanisms will be broken by this.